app/code/Magento/Inventory/Setup/Operation/CreateReservationTable.php

```diff
@@ -0,0 +1,127 @@
+use Magento\InventoryApi\Api\Data\ReservationInterface;
+use Magento\InventoryApi\Api\Data\StockInterface;
+
+class CreateReservationTable
+{
+    /**
+     * @param SchemaSetupInterface $setup
+     * @return void
+     */
+    public function execute(SchemaSetupInterface $setup)
+    {
+        $reservationTable = $this->createReservationTable($setup);
+
+        $setup->getConnection()->createTable($reservationTable);
+    }
+
+    /**
+     * @param SchemaSetupInterface $setup
+     * @return Table
+     */
+    private function createReservationTable(SchemaSetupInterface $setup)
+    {
+        $reservationTable = $setup->getTable(ReservationResourceModel::TABLE_NAME_RESERVATION);
+        $stockTable = $setup->getTable(StockResourceModel::TABLE_NAME_STOCK);
+
+        return $setup->getConnection()->newTable(
+            $reservationTable
+        )->setComment(
+            'Inventory Reservation Table'
+        )->addColumn(
+            ReservationInterface::RESERVATION_ID,
+            Table::TYPE_INTEGER,
+            null,
+            [
+                Table::OPTION_IDENTITY => true,
+                Table::OPTION_UNSIGNED => true,
+                Table::OPTION_NULLABLE => false,
+                Table::OPTION_PRIMARY => true,
+            ],
+            'Reservation ID'
+        )->addColumn(
+            ReservationInterface::STOCK_ID,
+            Table::TYPE_INTEGER,
+            null,
+            [
+                Table::OPTION_NULLABLE => false,
+                Table::OPTION_UNSIGNED => true,
+            ],
+            'Stock ID'
+        )->addColumn(
+            ReservationInterface::SKU,
+            Table::TYPE_TEXT,
+            64,
+            [
+                Table::OPTION_NULLABLE => false,
+            ],
+            'Sku'
+        )->addColumn(
+            ReservationInterface::QUANTITY,
+            Table::TYPE_DECIMAL,
+            null,
+            [
+                Table::OPTION_UNSIGNED => false,
+                Table::OPTION_NULLABLE => false,
+                Table::OPTION_DEFAULT => 0,
+            ],
+            'Quantity'
+        )->addColumn(
+            ReservationInterface::STATUS,
+            Table::TYPE_INTEGER,
+            null,
+            [
+                Table::OPTION_NULLABLE => false,
+                Table::OPTION_UNSIGNED => true,
+            ],
+            'Status'
+        )->addForeignKey(
+            $setup->getFkName(
+                $reservationTable,
+                ReservationInterface::STOCK_ID,
+                $stockTable,
+                StockInterface::STOCK_ID
+            ),
+            ReservationInterface::STOCK_ID,
+            $stockTable,
+            StockInterface::STOCK_ID,
+            AdapterInterface::FK_ACTION_CASCADE
+        )->addForeignKey(
+            $setup->getFkName(
+                $reservationTable,
+                ReservationInterface::SKU,
+                'catalog_product_entity',
```

Comment:
we are not allowed to make **Cross-Module Foreign Keys**,
because Database Structure could not be considered as Module's contract 
That's why Module could rely just on programming APIs (Service Layer) of another module, but not database tables. Because tables is a part of implementation which could be changed any time in future (for example by 3rd party developer).